model.py

```diff
@@ -328,13 +328,18 @@ class SessionManager(object):
     engine_for_url = {}
 
     @classmethod
-    def engine(cls, url=None):
+    def engine(cls, url=None, session=None):
         url = url or Configuration.database_url()
         return create_engine(url, echo=DEBUG)
 
     @classmethod
-    def sessionmaker(cls, url=None):
-        engine = cls.engine(url)
+    def sessionmaker(cls, url=None, session=None):
+        if not (url or session):
+            url = Configuration.database_url()
+        if url:
+            engine = cls.engine(url)
+        elif session:
```

Comment:
This addition allows a session factory to be created from an existing Session object.